### PR TITLE
feat(patterns): add Buttons docs

### PR DESCRIPTION
**Pull request by james-sann**
Added Buttons pattern documentation page with image examples.

## Checklist

- [ ] :link: considered opportunities for adding cross-reference URLs (grep for keywords)
- [ ] :wheelchair: analyzed via [axe](https://www.deque.com/axe/) and evaluated using VoiceOver
- [ ] :memo: tested in Chrome, Firefox, Safari, Edge, and IE11

### CODE REVIEW

File: src/pages/patterns/buttons.mdx [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+    <ul>
+      <li>Avoid mixing and matching button and input sizes.</li>
+    </ul>
+  </Dont>
+</BestPractice>
+
+### Alignment
+
+When a basic or default button is placed in conjunction with a primary button,
+the primary button should always be placed at the end in left-to-right layouts.
```
Comment (jzempel) [inline src/pages/patterns/buttons.mdx:48]:
```suggestion
the primary button should always be placed at the end.
```

Technically "the end" is always the end – whether in LTR or RTL :wink:.

File: src/pages/patterns/buttons.mdx [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+  caption="Arabic RTL example"
+  imgAlt="Arabic RTL example"
+  imageSource={props.data.buttonsLocalizationArabic.childFile.childImageSharp.gatsbyImageData}
+/>
+
+## Accessibility
+
+### Keyboard support
+
+Users must be able activate a button by pressing `Enter` or `Space` while the button has focus.
```
Comment (jzempel) [inline src/pages/patterns/buttons.mdx:231]:
```suggestion
Users must be able to activate a button by pressing `Enter` or `Space` while the button has focus.
```

File: src/pages/patterns/buttons.mdx (hunk 1 above)
Comment (james-sann) [reply src/pages/patterns/buttons.mdx:48]: Agreed @jzempel. I originally had it as your recommendation here but Jen added those extra words when she reviewed the Google Doc. Will keep it as I originally had it.

File: src/pages/patterns/buttons.mdx (hunk 2 above)
Comment (james-sann) [reply src/pages/patterns/buttons.mdx:231]: @jzempel Thank you for the extra eagle eyes.

**Review by jzempel — APPROVED** [review]

**Pull request merged**